src/commands/generate.ts

```diff
@@ -52,6 +52,53 @@ async function generate(toolbox: GluegunToolbox) {
     pascalName = pascalName.slice(0, -1 * pascalGenerator.length)
     command(`npx ignite-cli generate ${generator} ${pascalName}`)
   }
+  // Check if src/app exists, denoting an Expo Router app
+  if (generator === "screen") {
+    const isExpoRouterApp = filesystem.exists("src/app") === "dir"
```

Comment:
So this is the correct directory structure to determine expo-router in an Ignite application, however omitting the `src/` nesting is totally valid and it is possible a dir structure looks like this:

- app (the routes)
- components
- hooks
- etc...

This is in fact the default expo application structure and Ignite generators should continue to be compatible for any project, not just our opinionated structure.

So perhaps we should look into a different mechanism for determining this? Maybe we check `app` or `src/app` to cover both. Or look for `package.json` to have `expo-router`?

Reply:
Yeah, that definitely tracks. I think it would make sense to check for `expo-router`. Do you think it make more sense to check within the `package.json` dependencies or within the `app.json` expo plugins?

Reply:
I'd say `package.json` is easier because you know where it will live, under `dependencies`, so you can search for the key there.

Issues with the `app.json` route:
- It could also be in `app.config.js` or `app.config.ts`
- Plugins is an array of a mix of strings and/or arrary, making searching through it a bit more difficult

Reply:
Yeah that's fair, thanks!